class-components/src/store/store.ts

```diff
@@ -0,0 +1,10 @@
+import { SEARCH_KEY } from './store.constants';
+
+export class Store {
+  static setSearchValue(searchValue: string) {
+    localStorage.setItem(SEARCH_KEY, JSON.stringify(searchValue));
+  }
+  static getSearchValue() {
+    return JSON.parse(localStorage.getItem(SEARCH_KEY) || '[]');
+  }
+}
```

Comment:
Hmm, no, this is not how you are supposed to pass state value between siblings in React, not by global value. You cannot handle any updates with this approach. What you should do is create a parent for your search and main part, create a state for it with input value, and pass a function as params into search and value as params into main. This trick is known as [state lifting](https://legacy.reactjs.org/docs/lifting-state-up.html#gatsby-focus-wrapper) This will allow you to properly re-render components when it's necessary, and make search available by typing, not only by pressing the button, for example. if you need local storage for this, you can also handle it from this parent component, but you should only set value by this handle, and get value only on mount to set it into store.

Also, there is a cool thing for this type of problem - [context](https://legacy.reactjs.org/docs/context.html#gatsby-focus-wrapper). if you use it, you will not be forced to pass this input value through a whole set of components into one particular that requires it.

And also, there is redux, but you are prohibited from using it in this task)

All links leads to the legacy docs, because they has examples with class, and new one would be only with function components